react/lib/src/components/progressbar/progressbar.tsx

```diff
@@ -4,23 +4,21 @@ import {createProgressbar} from '@agnos-ui/react-headless/components/progressbar
 import {useWidgetWithConfig} from '@agnos-ui/react-headless/config';
 import {toSlotContextWidget} from '@agnos-ui/react-headless/types';
 import {Slot} from '@agnos-ui/react-headless/slot';
+import {useDirective} from '@agnos-ui/react-headless/utils/directive';
+import classNames from 'classnames';
 
 export * from '@agnos-ui/react-headless/components/progressbar';
 
 function DefaultSlotContent(slotContext: ProgressbarContext) {
-	const classList = ['progress-bar'];
-	if (slotContext.state.striped) {
-		classList.push('progress-bar-striped');
-	}
-	if (slotContext.state.animated) {
-		classList.push('progress-bar-animated');
-	}
-	if (slotContext.state.className) {
-		classList.push(slotContext.state.className);
-	}
+	const className = classNames(
+		'progress-bar',
+		{'progress-bar-striped': slotContext.state.striped},
```

Comment:
Maybe we can avoid the repetition of `slotContext.state` by extracting the variables first : `const {striped, animated, className} = slotContext.state;